### DISCUSSION TITLE
Typia: Typebox Alternative

### DISCUSSION THREAD

**Discussion by ali-abdelqawy** [elysiajs/elysia#957] [Ideas]
Would it be better if Typebox gets replaced with Typia?
https://typia.io/

**Comment by DIEGOHORVATTI** [comment]
@ali-abdelqawy Why would you prefer typia? Is there any limitation in typebox?

**Reply by ali-abdelqawy** [reply]
@DIEGOHORVATTI I like both of them, I don't prefer Typia over Typebox, I'm just curious if anyone has thoughts on that, what do you think?

The only thing that is noticeable for me is that Typia outperforms Typebox in every benchmark except the first one
https://github.com/samchon/typia/tree/master/benchmark/results/AMD%20Ryzen%209%207940HS%20w%20Radeon%20780M%20Graphics

**Reply by DIEGOHORVATTI** [reply]
> @DIEGOHORVATTI I like both of them, I don't prefer Typia over Typebox, I'm just curious if anyone has thoughts on that, what do you think?
> 
> The only thing that is noticeable for me is that Typia outperforms Typebox in every benchmark except the first one https://github.com/samchon/typia/tree/master/benchmark/results/AMD%20Ryzen%209%207940HS%20w%20Radeon%20780M%20Graphics

It would be interesting to use any validation manager to tell the truth, but elysia has several problems with the typebox linked to openApi, I think it's difficult for them to remove this now

https://github.com/elysiajs/elysia/discussions/642#discussioncomment-9462773

**Reply by DIEGOHORVATTI** [reply]
So until it's 100% stable, modifications like this are infeasible, my brother. @ali-abdul qawy right @diogojorge1401?
